### ISSUE TITLE
License approval

### ISSUE THREAD

**Issue by gnud** [lead-ratings/gender-guesser#6]
Hi, can I use this library commercially or make distributions?

**Comment by DZPM** [comment]
Hi @gnud , the library is Free Software, licensed under the GPL v3:
https://github.com/lead-ratings/gender-guesser/blob/master/LICENSE

Of course you can use it commercially (we do!), you just need to keep the GPL 3 licence, disclose the source code, publish any modifications, etc.

**Comment by Shruthiramesh93** [comment]
Hello @DZPM , I am trying to run the gender_guessor package on the large CSV data file, I tried a couple of times to run this package, but no luck. Could you share a doc, where I can refer to?

**Comment by gnud** [comment]
> Hi @gnud , the library is Free Software, licensed under the GPL v3:
> https://github.com/lead-ratings/gender-guesser/blob/master/LICENSE
> 
> Of course you can use it commercially (we do!), you just need to keep the GPL 3 licence, disclose the source code, publish any modifications, etc.

If I don't need to make modifications, do I still need to publish the source code or do I need to publically state somewhere that I use this library?
My problem is I don't know where to put the info, not that I mind disclosing the info.

So, let me rephrase the question, if I use the library internally, without making modifications, only imported in my project, do I need to disclose the source code of my project, or publish the source code of this project in my site publically?

**Comment by DZPM** [comment]
@Shruthiramesh93, please open a new Issue if you have new questions, don't ask in unrelated Issues.

The only doc is the README:
https://github.com/lead-ratings/gender-guesser/blob/master/README.rst

**Comment by DZPM** [comment]
@gnud (Disclaimer, I am not a lawyer!)

If you don't do modifications, you don't need to publish anything, as it's already published here.
If you are using it as an external library, you don't need to change the license of your project.
You only need to disclose that you're using `gender-guesser` by Lead Ratings, and link to this repository for the source code: https://github.com/lead-ratings/gender-guesser

Good luck in your project!

**Comment by gnud** [comment]
> @gnud (Disclaimer, I am not a lawyer!)
> 
> If you don't do modifications, you don't need to publish anything, as it's already published here.
> If you are using it as an external library, you don't need to change the license of your project.
> You only need to disclose that you're using `gender-guesser` by Lead Ratings, and link to this repository for the source code: https://github.com/lead-ratings/gender-guesser
> 
> Good luck in your project!

One more thing, do you know some platform that discloses the FOSS libraries they use? - Cause I need a template how to present them in my site :). Probably I guess legal section.

**Comment by DZPM** [comment]
Yes, a legal section would be ok.

As an example, check "Licenses and Acknowledgements for Incorporated Software":
https://docs.python.org/3/license.html#licenses-and-acknowledgements-for-incorporated-software

**Comment by tgamblin** [comment]
@DZPM:

> If you are using it as an external library, you don't need to change the license of your project.

This part, at least, is not true for GPLv3.  This is true for LGPL (v2 and v3). With GPLv3, the whole program using this library is be considered "derived work", and you need to distribute the source of the entire system.
